python/dolfinx/io/gmshio.py

```diff
@@ -167,9 +171,9 @@ def extract_geometry(model: gmsh.model, name: typing.Optional[str] = None) -> np
         return points[perm_sort]
 
     def model_to_mesh(model: gmsh.model, comm: _MPI.Comm, rank: int, gdim: int = 3,
-                      partitioner: typing.Callable[
-            [_MPI.Comm, int, int, AdjacencyList_int32], AdjacencyList_int32] =
-            create_cell_partitioner(GhostMode.none), dtype=default_real_type) -> typing.Tuple[
+                      partitioner: typing.Optional[typing.Callable[
+                          [_MPI.Comm, int, int, AdjacencyList_int32], AdjacencyList_int32]] = None,
+                      dtype=default_real_type) -> typing.Tuple[
```

Comment:
Why this change of partitioner?

Reply:
Consistent with change elsewhere, see #2727. It avoids calling a partitioner in serial case.